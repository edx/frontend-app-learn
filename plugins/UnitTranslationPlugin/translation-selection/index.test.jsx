import { shallow } from '@edx/react-unit-test-utils';

import TranslationSelection from './index';

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useContext: jest.fn().mockName('useContext').mockReturnValue({
    authenticatedUser: {
      userId: '123',
    },
  }),
}));
jest.mock('@edx/paragon', () => ({
  IconButton: 'IconButton',
  Icon: 'Icon',
  ProductTour: 'ProductTour',
}));
jest.mock('@edx/paragon/icons', () => ({
  Language: 'Language',
}));
jest.mock('./useTranslationTour', () => () => ({
  translationTour: {
    abitrarily: 'defined',
  },
  isOpen: false,
  open: jest.fn().mockName('open'),
  close: jest.fn().mockName('close'),
}));
jest.mock('react-redux', () => ({
  useDispatch: jest.fn().mockName('useDispatch'),
}));
jest.mock('@src/generic/plugin-store', () => ({
  registerOverrideMethod: jest.fn().mockName('registerOverrideMethod'),
}));
jest.mock('./TranslationModal', () => 'TranslationModal');
jest.mock('./useSelectLanguage', () => () => ({
  selectedLanguage: 'en',
  setSelectedLanguage: jest.fn().mockName('setSelectedLanguage'),
}));
jest.mock('../feedback-widget', () => 'FeedbackWidget');

describe('<TranslationSelection />', () => {
  const props = {
    id: 'plugin-test-id',
    courseId: 'course-v1:edX+DemoX+Demo_Course',
    language: 'en',
    availableLanguages: [
      {
        code: 'en',
        label: 'English',
      },
      {
        code: 'es',
        label: 'Spanish',
      },
    ],
    unitId: 'unit-test-id',
  };
  it('renders', () => {
    const wrapper = shallow(<TranslationSelection {...props} />);
    expect(wrapper.snapshot).toMatchSnapshot();
  });
});
