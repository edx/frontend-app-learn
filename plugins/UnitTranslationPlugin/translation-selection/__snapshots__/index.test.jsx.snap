// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TranslationSelection /> renders 1`] = `
<Fragment>
  <ProductTour
    tours={
      Array [
        Object {
          "abitrarily": "defined",
        },
      ]
    }
  />
  <IconButton
    alt="change-language"
    className="mr-2 mb-2 float-right"
    iconAs="Icon"
    id="translation-selection-button"
    onClick={[MockFunction open]}
    src="Language"
    variant="primary"
  />
  <TranslationModal
    availableLanguages={
      Array [
        Object {
          "code": "en",
          "label": "English",
        },
        Object {
          "code": "es",
          "label": "Spanish",
        },
      ]
    }
    close={[MockFunction close]}
    courseId="course-v1:edX+DemoX+Demo_Course"
    id="plugin-test-id"
    isOpen={false}
    selectedLanguage="en"
    setSelectedLanguage={[MockFunction setSelectedLanguage]}
  />
  <FeedbackWidget
    courseId="course-v1:edX+DemoX+Demo_Course"
    translationLanguage="en"
    unitId="unit-test-id"
    userId="123"
  />
</Fragment>
`;
